Added to ProductInfo:

jleiandy opened the pull request:
Star rating, product category, product name, product price,
social media buttons, and style selector

jleiandy committed:
Added to ProductInfo:
Star rating, product category, product name, product price,
social media buttons, and style selector

jleiandy committed:
Updated json to include moment npm module

cschucode commented on client/components/ProductOverview/ImageGallery.jsx:
```diff
@@ -36,8 +36,8 @@ const ImageGallery = ({ currentStyle, imageIndex, updateImageIndex }) => {
       <ThumbnailView photos={photos} handleClick={handleThumbNailClick} className="thumbnailComponent" currentIndex={imageIndex} />
-      <AiOutlineArrowLeft className="leftArrow" onClick={() => handleArrow('left')} size={arrowSize} />
-      <AiOutlineArrowRight className="rightArrow" onClick={() => handleArrow('right')} size={arrowSize} />
+      {imageIndex === 0 ? null : <AiOutlineArrowLeft className="leftArrow" onClick={() => handleArrow('left')} size={arrowSize} />}
```
I'm wondering if for these instead of a ternary would the `&&` work like ```imageIndex !== 0 && <AiOUtline... />```

jleiandy replied:
Perhaps, what would be the benefits of this?

cschucode replied:
Reduce complexity. You could get rid of `null` and if image index doesn't equal zero, then render arrow left.

cschucode commented on client/components/ProductOverview/ProductOverview.jsx:
```diff
@@ -23,23 +47,55 @@ const ProductOverview = ({ productId }) => {
+
+  if (currentStyle) {
+    return (
+      <div>
```
This is a really big return statement with a lot going on. I'm wondering if there's any way to break apart some of this into smaller components? Social media could be its own component etc. Also, you can use a ternary rather than the **if** statement like 
```
return !currentStyle ? <div>Loading</div> : <div>All the long component stuff code</div>;

jleiandy replied:
honestly, this condition return is a workaround. On mount the ProductOverview component needs to complete an axios call to get the currentstyle. Since that is async, the component will try to render before it has all of the information it needs and then it breaks. That's why I set an if statement to return a basic "Loading Product Info" div while the axios request is finishing. Not sure if this is considered hacky. Would like your thoughts.

cschucode commented on client/components/ProductOverview/StyleSelector.jsx:
```diff
@@ -0,0 +1,51 @@
+import '../../css/StyleSelector.scss';
+
+const StyleSelector = ({ styles, currentStyle, updateCurrentStyle }) => {
+  if (styles) {
```
I would use a ternary operator here as well. After looking at this, seems like both ways are acceptable. https://reactjs.org/docs/conditional-rendering.html

jleiandy replied:
> Looks good so far. Are the styles the same as the thumbnails? I thought they were different for some reason?

Styles are essentially a specific set of thumbnails. The style selector lets you choose between the different sets by providing the first thumbnail from each set to click on.

cschucode commented on client/components/ProductOverview/ThumbnailView.jsx:
```diff
@@ -19,6 +19,9 @@ const ThumbnailView = ({ photos, handleClick, currentIndex }) => {
         setThumbnailEnd(7);
+      } else {
+        setThumbnailStart(currentIndex);
+        setThumbnailEnd(currentIndex + 7);
```
Without knowing what were doing, it's hard to know what this 7 represents. A nice variable like `THUMBNAIL_UPPER_LIMIT` might help to clear things up.

jleiandy replied:
Made 7 into a constant.

cschucode commented on client/css/StyleSelector.scss:
```diff
@@ -0,0 +1,13 @@
+
+.styleThumbnail {
+  margin: 5px 1%;
+  height: 30%;
```
😢  percentages. 30% of what?

jleiandy replied:
Not sure why my height is a percentage, I think I was just testing different css options and forgot to change this when I commited. My new commit will fix this.

But I do use percentages for the width.
This is actually for my implementation of having only 4 thumbnails per row. I set the thumbnail container to be a flexbox row with flexwrap on, then I set the width of each thumbnail to be the total width of the container divided by 4.

cschucode commented on client/utils/getStarRatingAvg.js:
```diff
@@ -13,7 +13,7 @@ const getStarRatingAvg = (productId) => new Promise((resolve, reject) => {
       const avg = Math.round((count * 100) / results.length) / 100;
 
-      resolve(avg);
+      resolve({ avg, count });
```
Nice refactor! 👍  I approve of this 😉

cschucode reviewed (commented):
Looks good so far. Are the styles the same as the thumbnails? I thought they were different for some reason?

jleiandy committed:
Added expand function and started AddToCart function
expand and addtocart are WIP, note there is a bug in the
addtocart component with regards to the size selector not resetting
upon style change. Will work on that.
need to add zoom in feature to the expand feature.
Also made some refactors based on Chris' comments.

jleiandy committed:
Implemented zoom feature

jleiandy committed:
Merge branch 'main' of https://github.com/HRSF137-FEC-Enoki/catwalk into main

jleiandy committed:
Merge branch 'main' into ProductInfo
Handled merge conflicts

jleiandy committed:
Set images in thumbnail view to be backgrounds of divs
allowed for easier managing of image size and position

cschucode reviewed (approved):
Nice clean-up and use of smaller components! Starting to come together!!
